src/index.js

```diff
@@ -1447,5 +1452,266 @@ module.exports = {
+            score++;
+          }
+        }
+
+        if (style && element.hasAttribute('style')) {
+          const expectedStyles =
+            typeof style === 'string' ? styleStringToObject(style) : style;
+          const actualStyles = styleStringToObject(
+            element.getAttribute('style')
+          );
+
+          Object.keys(expectedStyles).forEach(styleName => {
+            const expectedStyle = expectedStyles[styleName];
+            const actualStyle = actualStyles[styleName];
+
+            if (actualStyle) {
+              score++;
+            }
+
+            if (isTextMatching(actualStyle, expectedStyle)) {
+              score++;
+            }
+          });
+        }
+
+        const specialAttributes = ['style', 'class'];
+        const ids = ['id', 'data-test-id', 'data-testid'];
+
+        Object.keys(attributes).forEach(attributeName => {
+          if (specialAttributes.indexOf(attributeName) !== -1) {
+            return; // skip
+          }
+
+          if (element.hasAttribute(attributeName)) {
+            if (typeof attributes[attributeName] === 'boolean') {
+              score++;
+            }
+
+            if (
+              element.getAttribute(attributeName) === attributes[attributeName]
+            ) {
+              score += ids.indexOf(attributeName) !== -1 ? 100 : 1;
+            }
+          } else if (!attributes[attributeName]) {
+            score++;
+          }
+        });
+      }
+
+      const expectedChildren = spec.children || [];
+
+      expectedChildren.forEach((childSpec, i) => {
+        const child = element.childNodes[i];
+        const childType = expect.findTypeOf(child);
+
+        if (typeof childSpec.nodeType === 'number') {
+          if (child && child.nodeType === childSpec.nodeType) {
+            if (childType.is('DOMElement')) {
+              // Element
+              score += scoreElementAgainstSpec(
+                element.childNodes[i],
+                convertDOMNodeToSatisfySpec(childSpec)
+              );
+            }
+
+            score++;
+          } else if (expect.findTypeOf(childSpec).is('DOMIgnoreComment')) {
+            score++;
+          }
+        } else if (
+          childType.is('DOMElement') &&
+          typeof childSpec === 'object'
+        ) {
+          score += scoreElementAgainstSpec(element.childNodes[i], childSpec);
+        } else if (
+          childType.is('DOMTextNode') &&
+          isTextMatching(child.nodeValue, childSpec)
+        ) {
+          score++;
+        }
+      });
+
+      return score;
+    }
+
+    function findMatchesWithGoodScore(data, spec) {
+      const elements =
+        typeof data.length === 'number' ? Array.from(data) : [data];
+
+      const result = [];
+      let bestScore = 0;
+
+      elements.forEach(element => {
+        const score = scoreElementAgainstSpec(element, spec);
+        bestScore = Math.max(score, bestScore);
+
+        if (score > 0 && score >= bestScore) {
```

Comment:
I think `score > 0` is redundant since `bestScore` is initialized to 0.

Reply:
I'm not sure it is, if the score is coming back as zero we want to disregard it, but if the score is the same as our best score and above zero we want to keep it.

Reply:
Ah! :+1: